EventGrid bindings conformance test is unreliable

<!-- If you need to report a security issue please visit https://docs.dapr.io/operations/support/support-security-issues -->

## Expected Behavior

`bindings.azure.eventgrid` conformance test should pass reliably when executed.

## Actual Behavior

`bindings.azure.eventgrid` is one of the common causes of the scheduled conformance test failures:

* https://github.com/dapr/components-contrib/actions/runs/1011465074
* https://github.com/dapr/components-contrib/runs/3018552393?check_suite_focus=true

The bindings conformance tests relies on a `sleep` to try to ensure that the `Invoke()` to publish a message happens after the  `Read()` subscription is set up. The current sleep duration is not currently long enough for the bindings.azure.eventgrid test to avoid publishing the test event before the subscription has successfully created, and so occasionally results in the test failure:

```
Running tests for TestBindingsConformance/bindings/azure.eventgrid ... 
  === RUN   TestBindingsConformance
  === RUN   TestBindingsConformance/azure.eventgrid
  === RUN   TestBindingsConformance/azure.eventgrid/init
  === RUN   TestBindingsConformance/azure.eventgrid/operations
  === RUN   TestBindingsConformance/azure.eventgrid/read
  time="2021-07-08T11:34:01.647084889Z" level=debug msg="Attempting to create or update Event Grid subscription. scope=*** endpointURL=https://cfbab24a9c99.ngrok.io/api/events" instance=fv-az180-858 scope=testLogger type=log ver=unknown
  === RUN   TestBindingsConformance/azure.eventgrid/create
  time="2021-07-08T11:34:02.966687745Z" level=debug msg="Successfully posted event to ***" instance=fv-az180-858 scope=testLogger type=log ver=unknown
  === RUN   TestBindingsConformance/azure.eventgrid/verify_read
  time="2021-07-08T11:34:03.202371626Z" level=debug msg="Succeeded to create or update Event Grid subscription. scope=*** endpointURL=https://cfbab24a9c99.ngrok.io/api/events" instance=fv-az180-858 scope=testLogger type=log ver=unknown
  time="2021-07-08T11:34:03.20242423Z" level=debug msg="About to start listening for Event Grid events at http://localhost:9000/api/events" instance=fv-az180-858 scope=testLogger type=log ver=unknown
      assertion_compare.go:313: 
        	  Error Trace:	bindings.go:208
        	  Error:      	"0" is not greater than "0"
        	  Test:       	TestBindingsConformance/azure.eventgrid/verify_read
        	  Messages:   	[]
  --- FAIL: TestBindingsConformance (241.32s)
      --- FAIL: TestBindingsConformance/azure.eventgrid (241.32s)
          --- PASS: TestBindingsConformance/azure.eventgrid/init (0.00s)
          --- PASS: TestBindingsConformance/azure.eventgrid/operations (0.00s)
          --- PASS: TestBindingsConformance/azure.eventgrid/read (0.00s)
          --- PASS: TestBindingsConformance/azure.eventgrid/create (0.32s)
          --- FAIL: TestBindingsConformance/azure.eventgrid/verify_read (240.00s)
  FAIL
  FAIL	github.com/dapr/components-contrib/tests/conformance	241.342s
  FAIL
  Completed tests for TestBindingsConformance/bindings/azure.eventgrid ... 
  Setting CONFORMANCE_FAILURE
```

Ideally, this should be synchronized to avoid the use of `sleep`, but the [comment in bindings.go](https://github.com/dapr/components-contrib/blob/0777a6a943548052d685d80d7e029ae0be8b8911/tests/conformance/bindings/bindings.go#L158) suggests that there does not exist a mechanism today to be informed when the subscription has actually completed. In the absence of such a mechanism, the sleep duration should be increased to reduce the incidence rate of this failure. 

## Steps to Reproduce the Problem

Run the conformance.yml GitHub workflow repeatedly. 

## Release Note

RELEASE NOTE: N/A

--- a/tests/config/bindings/tests.yml
+++ b/tests/config/bindings/tests.yml
@@ -2,6 +2,7 @@
 # Config map:
 ## output: A map of strings that will be part of the request for the output binding
 ## readBindingTimeout : timeout to wait to receive test event
+## readBindingWait : duration to wait after successful Read before attempting to Publish
 ## url: specific to http component, url of the http server
 ## method: specific to http component, what method to use
 componentType: bindings
@@ -22,6 +23,7 @@ components:
     operations: ["create", "operations", "read"]
     config:
       readBindingTimeout: 240s
+      readBindingWait: 8s
   - component: azure.storagequeues
     operations: ["create", "operations", "read"]
   - component: azure.servicebusqueues

--- a/tests/conformance/bindings/bindings.go
+++ b/tests/conformance/bindings/bindings.go
@@ -21,6 +21,7 @@ import (
 
 const (
 	defaultTimeoutDuration = 60 * time.Second
+	defaultWaitDuration    = time.Second
 )
 
 // nolint:gochecknoglobals
@@ -38,6 +39,7 @@ type TestConfig struct {
 	InputMetadata      map[string]string `mapstructure:"input"`
 	OutputMetadata     map[string]string `mapstructure:"output"`
 	ReadBindingTimeout time.Duration     `mapstructure:"readBindingTimeout"`
+	ReadBindingWait    time.Duration     `mapstructure:"readBindingWait"`
 }
 
 func NewTestConfig(name string, allOperations bool, operations []string, configMap map[string]interface{}) (TestConfig, error) {
@@ -52,6 +54,7 @@ func NewTestConfig(name string, allOperations bool, operations []string, configM
 		InputMetadata:      make(map[string]string),
 		OutputMetadata:     make(map[string]string),
 		ReadBindingTimeout: defaultTimeoutDuration,
+		ReadBindingWait:    defaultWaitDuration,
 	}
 
 	err := config.Decode(configMap, &testConfig)
@@ -82,7 +85,7 @@ func startHTTPServer(url string) {
 }
 
 func (tc *TestConfig) createInvokeRequest() bindings.InvokeRequest {
-	// There is a possiblity that the metadata map might be modified by the Invoke function(eg: azure blobstorage).
+	// There is a possibility that the metadata map might be modified by the Invoke function(eg: azure blobstorage).
 	// So we are making a copy of the config metadata map and setting the Metadata field before each request
 	// Use CloudEvent as data because it is required by Azure's EventGrid.
 	cloudEvent := "[{\"eventType\":\"test\",\"eventTime\": \"2018-01-25T22:12:19.4556811Z\",\"subject\":\"dapr-conf-tests\",\"id\":\"A234-1234-1234\",\"data\":\"root/>\"}]"
@@ -159,7 +162,7 @@ func ConformanceTests(t *testing.T, props map[string]string, inputBinding bindin
 		// Need a small wait here because with brokers like MQTT
 		// if you publish before there is a consumer, the message is thrown out
 		// Currently, there is no way to know when Read is successfully subscribed.
-		time.Sleep(1000 * time.Millisecond)
+		time.Sleep(config.ReadBindingWait)
 	}
 
 	// CreateOperation

